documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,295 @@
+
+
+### Authentication Request
+
+CIBA allows the client to use login_hint_token, id_token_hint or login_hint as a hint in the authentication request. This CAMARA profile makes the login_hint parameter REQUIRED. The client SHALL specify login_hint (and only login_hint) in the authentication request when using CIBA in a CAMARA context.
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## Format of `login_hint`
+
+This CAMARA document clarifies the values used in login_hint in the following way:
+
+    * **_tel_**
+
+      For phone numbers. The `login_hint` must be a tel URI as defined in [RFC 3966](https://www.rfc-editor.org/info/rfc3966) for global phone numbers without visual separators in [E.164](https://www.itu.int/rec/T-REC-E.164-201011-I/en) format. For example, `tel:+34666666666`.
+
+    * **_ipport_**
+
+      For IPv4 and IPv6 addresses, that can optionally include a port. For example, `ipport:80.90.34.2:16790`, `ipport:80.90.34.2`, `ipport:[2001:db8::1]:8080` or `ipport:[2001:db8::1]`.
+
+
+## Offline Access
+
+#### Refresh Token Issuance
+
+Neither OIDC, CIBA, nor OAuth2 define a way for clients to indicate whether they need a refresh_token. 
+Refresh token issuance is optional and at the discretion of the AZ.
+
+Camara uses the scope `offline_access` in the authorization request to indicate to the AZ that the client requests a refresh token additionally to the access token for Camara API access.
+
+---
+**NOTE**
+
+OIDC defines the scope `offline_access` and it is used to get an access token to be used at the UserInfo endpoint, and also to obtain a request token to refresh this access token.
+
+---
+
+The OIDC rules for `offline_access` as defined in [OIDC Offline Access](https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccess) apply. 
+
+The OIDC section on [OIDC Using Refresh Tokens](https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokens) is not changed by this document.
+
+The OIDC security and privacy considerations regarding offline access and refresh tokens apply e.g. on
+[Token lifetime](https://openid.net/specs/openid-connect-core-1_0.html#TokenLifetime) and [Offline Access Privacy](https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccessPrivacy).
+
+### Refresh Token Usage
+
+This section applies to OIDC Authorization Code Flow and CIBA. Other flows do not need refresh tokens and MUST not use refresh tokens.
+
+In addition to [OIDC Using Refresh Tokens](https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokens) and [OAuth2 Refreshing an Access Token](https://datatracker.ietf.org/doc/html/rfc6749#section-6) this document defines the following:
+
+     * A new access token MUST not be issued if the user has revoked their consent.
+     * A new access token MUST not be issued if the client status regarding the requested API access has changed.
```

Comment:
```suggestion
```

Reply:
Both statements are not very technical and both prepare the implementer for cases where new access tokens must not be issued. I suggest to leave text unchanged.

Reply:
Can you give me an example scenario where "A new access token MUST not be issued if the client status regarding the requested API access has changed." occurs? Your previous answer did not clarify..

Reply:
The client was hacked. 
Tokens were leaked or were stolen and sender-contraint access tokens are not implemented.
The client terminated or changed their contract.
The client lied.

All not very technical and we do not have all the management APIs in place to automate this.
But the same is true, for consent management APIs.

Can we leave this text unchanged for now and merge, or do you see the _need_ for changes?

Reply:
> The term `client` in all documents is defined by OAuth2 / OIDC and is the API consumer.

Based on the OAuth2 specifications, the term 'client' refers to an application that makes protected resource requests on behalf of the resource owner and with its authorization. 

The last two example scenarios are not related to the client but rather to the end-user (resource owner).

So my recommendation remains the same.

Reply:
> > The term `client` in all documents is defined by OAuth2 / OIDC and is the API consumer.
> 
> Based on the OAuth2 specifications, the term 'client' refers to an application that makes protected resource requests on behalf of the resource owner and with its authorization.

I should not have written "API consumer" because that is just a SW component. https://github.com/camaraproject/Commonalities/blob/main/documentation/Glossary.md
 
`client` in my book is that entity that is onboarded and with which "we" have a contract which allows API access, quotas, scopes and whatnot.

> 
> The last two example scenarios are not related to the client but rather to the end-user (resource owner).
> 

That "client" changed their contract e.g. removing API1 from their contract.
That "client" lied during onboarding and the contract they have is void.

Why do you think that is the end-user (resource owner).

> So my recommendation remains the same.

Reply:
If we want to comply with the OAuth2 specifications conforming to RFC https://datatracker.ietf.org/doc/html/rfc6749, we should keep in mind that a client can act on behalf of an entity (either a company or an individual), but it is not the entity itself.

Revised scenarios aligned with the standard are as follows:

- A person (the resource owner) changes their contract, e.g., removes API1 (the resource) from their contract (granted authorization).
- A person lied during onboarding, and the contract (granted authorization) is declared void by the company (the other resource owner).

Reply:
A resource owner/end-user does not have a contract with "us". So, the AZ cannot know whether the contract between the end-user and the client (entity behind the client) has changed.

I would like to resolve this conversation if nobody chimes in.